File: leetcode/01-two-sum.py
from typing import List, Dict


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_to_idx: Dict[int, int] = {}
        for i in range(len(nums)):
            rest = target - nums[i]
            if rest in num_to_idx:
                return [i, num_to_idx[rest]]

            num_to_idx[nums[i]] = i

        return []

File: leetcode/0001/01-two-sum.cpp
#include <unordered_map>
#include <vector>

class Solution {
 public:
  std::vector<int> twoSum(std::vector<int>& nums, int target) {
    std::unordered_map<int, int> num_to_index{};
    for (auto i{0}; i < nums.size(); ++i) {
      if (num_to_index.contains(target - nums[i])) {
        return {i, num_to_index[target - nums[i]]};
      }
      num_to_index.insert({nums[i], i});
    }

    return {};
  }
};
